src/client/dfuse/pil4dfs/hook.c

```diff
@@ -892,16 +994,18 @@ register_a_hook(const char *module_name, const char *func_name, const void *new_
 
 	/* Do initialization work at the very first time. */
 	if (!num_hook) {
-		module_list = malloc(sizeof(struct module_patch_info_t) * MAX_MODULE);
+		D_ALLOC_ARRAY(module_list, MAX_MODULE);
 		if (module_list == NULL) {
-			printf("Failed to allocate memory for module_list.\nQuit\n");
+			D_FATAL("Failed to allocate memory for module_list. %d (%s)\n", ENOMEM,
+				strerror(ENOMEM));
```

Comment:
FYI - the D_ALLOC* calls will have loudly logged the failure anyway, there's no need to repeat the details here.